Show legends for each connected dataset in XY Plot

bgoldowsky opened the pull request:
For PT-186266524

Pulldown menus for X and Y attributes of all datasets should be correctly shown.  [X] button unlinks the dataset from the graph.

Known limitations, to be addressed in later stories:
* The actual graph data is still only shown for the first connected dataset.
* Removing the X attribute, or all of the Y attributes for any dataset, is not supported and may lead to errors or erroneous displays.
* Removing the actually-displayed dataset when there is more than one may result in erroneous display.

kswenson reviewed (changes requested):
👍 In general this looks good. My main suggestion is rather than replacing use of `useDataConfigurationContext` with explicit `layer` props, use the existing context to provide the appropriate layer-specific context. In this PR, that would involve assuming layer 0 in `graph.tsx` where the context is being provided now, but once multiple layers are supported it just means providing the context for each layer.

bgoldowsky commented:
Thank you @kswenson , I followed your suggestion and added DataConfiguration context providers inside the loops that create the legend.  It allowed me to leave several components unmodified that previously had to have arguments passed in, so I think it's a good change.

The other PR, #2063, will need updates to work with this so I've moved it back to draft, but this one can be re-reviewed.

kswenson reviewed (approved):
👍 Changes look good

tejal-shah reviewed (changes requested):
@bgoldowsky , this looks good from a functionality perspective. I just have some points from a styling angle for cleaning up:
1. There is no spacing between rows of series causing the top and bottom legend dropdowns to be struck to each other. The same is the case with the Add Series button - it gets stuck to the legend dropdown directly above it.
2. If an attribute name is a little long, it overflows outside the dropdown. It might be a design question, but IMHO maybe we could just clip the attribute name if we don't want to expand the dropdown, it might look better.
<img width="852" alt="Screenshot 2023-12-04 at 1 12 39 PM" src="https://github.com/concord-consortium/collaborative-learning/assets/81762587/407f52d2-fa7b-458a-90f8-3b325ea84fd0">
<img width="854" alt="Screenshot 2023-12-04 at 1 12 53 PM" src="https://github.com/concord-consortium/collaborative-learning/assets/81762587/8d8511eb-e29a-493d-8486-7e48234c7984">


Another question I have is that currently, there are a few deviations from the zeplin spec. For example, the Series Close button does not match the zeplin spec's hover or select states. Do you know if this would be a separate story or is this something we could do in this PR?

tejal-shah commented:
@bgoldowsky , thank you. This looks great. One last thing that's styling related (sorry) - The bottom padding between the dropdowns and the multilegend box border doesn't seem to be even for all multilegend boxes. The padding is a lot less for the last box.
<img width="854" alt="Screenshot 2023-12-04 at 3 47 57 PM" src="https://github.com/concord-consortium/collaborative-learning/assets/81762587/a61690c6-dc9a-4051-a42e-b82e50637b72">

And I also noticed that if you remove the legend for the first dataset that is currently plotted by using the close button, we automatically show/plot the other series that wasn't being plotted until now. But we only show it's connecting lines and not the actual points (unless the user reloads/the graph is redrawn). I can create a separate bug for it if you think it should be its own story. Screencast - ttps://screenpal.com/watch/c0lfYgVmtT6

bgoldowsky commented:
I increased the padding on the bottom.

Regarding the missing points - that is a separate issue. I think it ought to be fixed by PR #2063 .

tejal-shah commented:
Thank you @bgoldowsky , this looks good.
